util/syn_yosys.sh

```diff
@@ -18,31 +18,46 @@
 # use fusesoc to generate files and file list
 #-------------------------------------------------------------------------
 \rm -Rf build
-fusesoc --cores-root .. run --target=sim --setup --build formal > /dev/null 2>&1
+fusesoc --cores-root .. run --target=syn --setup lowrisc:systems:top_earlgrey
 
 # copy all files into directory "syn_out"
 \rm -Rf syn_out
 mkdir syn_out
-cp build/formal_0/src/*/*/*.sv build/formal_0/src/*/*/*/*.sv syn_out
+cp \
+  build/lowrisc_systems_top_earlgrey_0.1/src/*/*.sv \
+  build/lowrisc_systems_top_earlgrey_0.1/src/*/*/*.sv \
+  build/lowrisc_systems_top_earlgrey_0.1/src/*/*/*/*.sv \
+  syn_out
+cd syn_out
+
+# not synthesizable
+\rm prim_pad_wrapper.sv
+\rm prim_generic_pad_wrapper.sv
+
+# not used for synthesis
+\rm pins_if.sv
+\rm tlul_assert.sv
+\rm tlul_assert_multiple.sv
+
+# match filename to module name
+mv ibex_register_file{_ff,}.sv
+
+# prim_util_memload.sv is only meant to be included within a module
+mv prim_util_memload.sv{,h}
+sed -i.delete.bak -e "s/prim_util_memload\.sv/prim_util_memload.svh/" *.sv
+rm *.delete.bak  # some sed implementations require backup extensions
 
 #-------------------------------------------------------------------------
 # convert all RTL files to Verilog
 #-------------------------------------------------------------------------
-cd syn_out
-
-for file in *.sv; do
-  module=`basename -s .sv $file`
-  echo $file
-  sv2v --define=VERILATOR --define=SYNTHESIS *_pkg.sv prim_assert.sv $file > ${module}.v
-
-  # TODO: eventually remove below hack. It removes "unsigned" from params
-  # because Yosys doesn't support unsigned parameters
-  sed -i 's/parameter unsigned/parameter/g' ${module}.v
-  sed -i 's/localparam unsigned/localparam/g' ${module}.v
+printf "\n\nsv2v:\n"
+sv2v -DSYNTHESIS *.sv > combined.v
+# split files up
+modules=`cat combined.v | grep "^module" | sed -e "s/^module //" | sed -e "s/ (//"`
+for module in $modules; do
+  cat combined.v | sed -n "/^module $module /,/^endmodule/p" > $module.v
```

Comment:
You could write this as `sed <combined.v >$module.v blah blah` to avoid the extra `cat`. Also, I'd consider writing the list of modules to a file (and then reading it back in to the modules variable). This code looks a bit fragile, and it would probably be easier to debug that way if something goes wrong.